# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/base/pickle.cc
# Commit: c02fa459e77d
# Full Hash: c02fa459e77d2572b2fe740f20d6070a77383589
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 72dc6537cf0b -r c02fa459e77d ipc/chromium/src/base/pickle.cc
--- a/ipc/chromium/src/base/pickle.cc	Mon Jun 21 21:53:11 2021 +0000
+++ b/ipc/chromium/src/base/pickle.cc	Mon Jun 21 21:53:11 2021 +0000
@@ -471,6 +471,11 @@
   }
 }
 
+void Pickle::Truncate(PickleIterator* iter) {
+  size_t dropped = buffers_.Truncate(iter->iter_);
+  header_->payload_size -= dropped;
+}
+
 void Pickle::BeginWrite(uint32_t length, uint32_t alignment) {
   DCHECK(alignment % 4 == 0) << "Must be at least 32-bit aligned!";
 